pkg/api/api_default_service.go

```diff
@@ -143,6 +143,33 @@ func (cfm *CfmApiService) AppliancesPost(ctx context.Context, credentials openap
 	return openapi.Response(http.StatusCreated, a), nil
 }
 
+// AppliancesUpdateById -
+func (cfm *CfmApiService) AppliancesUpdateById(ctx context.Context, applianceId string, newApplianceId string) (openapi.ImplResponse, error) {
+	// Make sure the newApplianceId doesn't exist
+	_, exist := manager.GetApplianceById(ctx, newApplianceId)
+	if exist == nil {
+		err := common.RequestError{
+			StatusCode: common.StatusApplianceIdDuplicate,
+			Err:        fmt.Errorf("the new name (%s) already exists", newApplianceId),
+		}
+		return formatErrorResp(ctx, &err)
+	}
+
+	// Make sure the appliance exists
+	appliance, err := manager.GetApplianceById(ctx, applianceId)
```

Comment:
There's no need to call the same function twice (Appliance or Host).
If you want the unique error message on line 153, just:
```
        appliance, err := manager.GetApplianceById(ctx, applianceId)
	if err != nil {
		if err.StatusCode == StatusApplianceIdDuplicate {
			err := common.RequestError{
				StatusCode: common.StatusApplianceIdDuplicate,
				Err:        fmt.Errorf("the new name (%s) already exists", newApplianceId),
			}
			return formatErrorResp(ctx, &err)
		}
		return formatErrorResp(ctx, err.(*common.RequestError))
	}
```